Code/Tools/ProjectManager/Platform/Mac/O3DE_SDK_Launcher.cpp

```diff
@@ -0,0 +1,69 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+#include <AzCore/Component/ComponentApplication.h>
+#include <AzCore/Memory/SystemAllocator.h>
+#include <AzCore/Settings/SettingsRegistryMergeUtils.h>
+#include <AzCore/Utils/Utils.h>
+#include <AzFramework/Process/ProcessWatcher.h>
+
+#include <cstdlib>
+#include <mach-o/dyld.h>
+
+int main(int argc, char* argv[])
+{
+    // We need to pass in the engine path since we won't be able to find it by searching upwards.
+    // We can't use any containers that use our custom allocator till after the call to ComponentApplication::Create()
+    char enginePathParam[AZ_MAX_PATH_LEN];
+    char processPathStr[AZ_MAX_PATH_LEN];
+    uint32_t pathLen = AZ_MAX_PATH_LEN;
+    _NSGetExecutablePath(processPathStr, &pathLen);
+    snprintf(enginePathParam, AZ_MAX_PATH_LEN, "--engine-path=%s/../Engine", processPathStr);
+
+    char *commandLineParams[AZ_MAX_PATH_LEN] = {processPathStr, enginePathParam, nullptr};
+    
+    // Create a ComponentApplication to initialize the AZ::SystemAllocator and initialize the SettingsRegistry
+    AZ::ComponentApplication application(2, commandLineParams);
+    application.Create(AZ::ComponentApplication::Descriptor());
+
+    AZ::IO::FixedMaxPath processPath = AZ::IO::PathView(AZ::Utils::GetExecutableDirectory());
+    AZ::IO::FixedMaxPath enginePath = processPath/".."/"Engine";
+ 
+    AZ::IO::FixedMaxPath installedBinariesFolder;
+    if (auto settingsRegistry = AZ::SettingsRegistry::Get(); settingsRegistry != nullptr)
+    {
+        if (settingsRegistry->Get(installedBinariesFolder.Native(), AZ::SettingsRegistryMergeUtils::FilePathKey_InstalledBinaryFolder))
+        {
+            installedBinariesFolder = enginePath / installedBinariesFolder;
+        }
+    }
+    
+    // Install python packages that are required before launching ProjectManager.
+    AZ::IO::FixedMaxPath pythonPath = enginePath/"python"/"runtime"/"python-3.7.10-rev1-darwin"/"Python.framework"/"Versions"/"3.7"/"bin"/"python3";
```

Comment:
Forgot about this. We need a way to tell which python package version we're using. This is hardcoded to "python-3.7.10-rev1-darwin".

Reply:
Also you can acombine the string portions of the path into one string
```suggestion
    AZ::IO::FixedMaxPath pythonPath = enginePath / "python/runtime/python-3.7.10-rev1-darwin/Python.framework/Versions/3.7/bin/python3";
```

That being said if we used CMake here, we don't have to worry about determining the version of python to invoke

Reply:
We also have the `python.sh` script that would invoke the correct python.

Reply:
I tried the python.sh. But we can't invoke a shell script when launched from Finder(whether it's invoked by execve() or  or system()). It worked if I launched it from Terminal. It doesn't look like a shell is available if launching from Finder. I think CMake might work though. We can make the assumption that cmake is installed at /Applications/CMake.app/Contents/MacOS. I can try doing that.

Reply:
+1 to finding a way to remove these inline paths/commands, or at least minimize them, in favor of the existing shell scripts

Reply:
I was able to get the get_python.sh shell script to run with some help from @AMZN-ScottR . I updated the script to install o3de python package as well. Earlier, it was being done only in LYPython.cmake.